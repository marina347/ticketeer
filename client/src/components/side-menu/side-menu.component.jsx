import React from "react";
import UserIconSmall from "../user-icon/user-icon-small/user-icon-small.component";
import "./side-menu.styles.scss";

export const SideMenu = ({ currentUser }) => {
  return (
    <div id="side_menu" className="side-menu">
      <UserIconSmall
        url={currentUser.picture}
        name={currentUser.name}
        email={currentUser.email}
      />
    </div>
  );
};

export default SideMenu;
