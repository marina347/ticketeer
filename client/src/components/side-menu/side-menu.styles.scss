.side-menu {
  display: flex;
  justify-content: flex-end;
  align-items: center;
  flex: 0 0 50%;
  text-align: center;
}
