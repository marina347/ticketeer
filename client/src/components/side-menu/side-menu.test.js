import React from "react";
import { shallow } from "enzyme";
import "jest-styled-components";
import { SideMenu } from "./side-menu.component";

describe("SideMenu component", () => {
  let wrapper;
  let currentUser;

  beforeEach(() => {
    currentUser = {
      picture: "test/picture",
      email: "test@test.com",
      name: "John",
    };
    wrapper = shallow(<SideMenu currentUser={currentUser} />);
  });

  it("Should render SideMenu component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("Should propagate props to child component", () => {
    const childComponent = wrapper.find('[id="side_menu"]').childAt(0);
    expect(childComponent.prop("email")).toBe(currentUser.email);
    expect(childComponent.prop("name")).toBe(currentUser.name);
    expect(childComponent.prop("url")).toBe(currentUser.picture);
  });
});
