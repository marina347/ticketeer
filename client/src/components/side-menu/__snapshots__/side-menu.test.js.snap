// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideMenu component Should render SideMenu component 1`] = `
<div
  className="side-menu"
  id="side_menu"
>
  <UserIconSmall
    email="test@test.com"
    name="John"
    url="test/picture"
  />
</div>
`;
